include/lsToDiskMesh.hpp

```diff
@@ -98,26 +108,52 @@ template <class T, int D, class N = T> class lsToDiskMesh {
 
     values.reserve(normalVectors.size());
     normals.reserve(normalVectors.size());
+    materialIds.reserve(normalVectors.size());
 
     const bool buildTranslatorFlag = buildTranslator;
     unsigned long counter = 0;
     if (buildTranslatorFlag) {
       translator->reserve(normalVectors.size());
     }
 
-    for (hrleConstSparseIterator<hrleDomainType> it(levelSet->getDomain());
-         !it.isFinished(); ++it) {
-      if (!it.isDefined() || std::abs(it.getValue()) > maxValue) {
+    // an iterator for each levelset
+    std::vector<hrleConstSparseIterator<hrleDomainType>> iterators;
+    for (const auto levelSet : levelSets) {
+      iterators.push_back(
+          hrleConstSparseIterator<hrleDomainType>(levelSet->getDomain()));
+    }
+
```

Comment:
Might be a good idea to alias iterators.back() here for readability.
Something like
auto &topIt = iterators.back()